src/txmempool.cpp

```diff
@@ -1048,7 +1051,7 @@ void CTxMemPool::TrimToSize(size_t sizelimit, std::vector<uint256>* pvNoSpendsRe
                         continue;
                     auto iter = mapNextTx.lower_bound(COutPoint(txin.prevout.hash, 0));
                     if (iter == mapNextTx.end() || iter->first->hash != txin.prevout.hash)
-                        pvNoSpendsRemaining->push_back(txin.prevout.hash);
+                        pvNoSpendsRemaining->push_back(txin.prevout);
```

Comment:
Shouldn't this be including all outpoints that are no longer spent, rather than just outpoints from transactions where none of the outpoints are spent?